modules/social_features/social_group/src/SocialGroupHelperService.php

```diff
@@ -285,40 +285,28 @@ public function countGroupMembershipsForUser(string $uid): int {
   /**
    * {@inheritdoc}
    */
-  public function getGroupsToAddUrl(AccountInterface $account) {
-    $found = FALSE;
-    $accessible_group_type = NULL;
-
-    /** @var array $group_types */
-    $group_types = $this->entityTypeManager->getStorage('group_type')
+  public function getGroupsToAddUrl(AccountInterface $account, string $group_type = 'flexible_group'): Url|null {
+    /** @var array $group_type */
+    $group_type = $this->entityTypeManager->getStorage('group_type')
       ->getQuery()
       ->accessCheck()
+      ->condition('id', $group_type)
       ->execute();
 
-    // Get all available group types.
-    foreach ($group_types as $group_type) {
-      // When the user has permission to create a group of the current type, add
-      // this to the creation group array.
-      if ($account->hasPermission('create ' . $group_type . ' group')) {
-        if ($accessible_group_type === NULL) {
-          $accessible_group_type = $group_type;
-        }
-        else {
-          $found = TRUE;
-          break;
-        }
-      }
+    $group_type = reset($group_type);
+
+    // Verify that group type exist and loaded.
+    if (!$group_type) {
+      return NULL;
     }
 
-    // There's just one group this user can create.
-    if (!$found && isset($accessible_group_type)) {
-      // When there is only one group allowed, add create the url to create a
-      // group of this type.
+    // When the user has permission to create a group of the current type, add
+    // this to the creation group array.
+    if ($account->hasPermission('create ' . $group_type . ' group')) {
       return Url::fromRoute('entity.group.add_form', [
-        'group_type' => $accessible_group_type,
+        'group_type' => $group_type,
       ]);
     }
-
     return NULL;
   }
```

Comment:
I think we don't need to change this function and can instead deprecate it and remove it in the future. We probably don't want to change how it functions in case other distribution users are using it.

Within our own code-base we have only 1 usage of it which is a block that contains an add button, where we can probably also fix the issue without calling this method. The deprecation will help us spot that during the upgrade path of the product :)